### DISCUSSION TITLE
How to open miniapp?

### DISCUSSION THREAD

**Discussion by nodirnasirov** [telegraf/telegraf#1928] [Q&A]
I have setup a miniapp in telegram and configured menu button to launch it, but, I need to be able to send a user a message with a button, so that user can click on it and launch webapp. How can it be done?

**Comment by hypernova7** [comment]
Use [direct link](https://core.telegram.org/bots/webapps#direct-link-mini-apps) or a webapp button

[Show webapp button instead of commands button](https://core.telegram.org/bots/api#setchatmenubutton)
[Direct link with answerInlineQuery](https://core.telegram.org/bots/api#answerinlinequery)
[Inline keyboard](https://core.telegram.org/bots/api#inlinekeyboardmarkup)
[Keyboard](https://core.telegram.org/bots/api#inlinekeyboardmarkup)

Note: webapp buttons are only for private chats, use a url button with direct link for any chat

**Comment by MKRhere** [comment]
Have a look at the detailed docs for Mini Apps with different examples:
https://github.com/feathers-studio/telegraf-docs/tree/master/examples/mini-apps
